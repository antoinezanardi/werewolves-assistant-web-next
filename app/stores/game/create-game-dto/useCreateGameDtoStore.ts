import { defineStore } from "pinia";
import { ref } from "vue";
import { DEFAULT_GAME_OPTIONS } from "~/composables/api/game/constants/game-options/game-options.constants";
import type { CreateGameAdditionalCardDto } from "~/composables/api/game/dto/create-game/create-game-additional-card/create-game-additional-card.dto";

import { CreateGamePlayerDto } from "~/composables/api/game/dto/create-game/create-game-player/create-game-player.dto";
import { CreateGameDto } from "~/composables/api/game/dto/create-game/create-game.dto";
import type { GameAdditionalCardRecipientRoleName } from "~/composables/api/game/types/game-additional-card/game-additional-card.types";
import { GameOptions } from "~/composables/api/game/types/game-options/game-options.class";
import { ADDITIONAL_CARDS_DEPENDANT_ROLES } from "~/composables/api/role/constants/role.constants";
import type { RoleName } from "~/composables/api/role/types/role.types";
import { StoreIds } from "~/stores/enums/store.enum";
import { useRolesStore } from "~/stores/role/useRolesStore";
import { useLocalStorage } from "@vueuse/core";
import { LocalStorageKeys } from "~/utils/enums/local-storage.enums";

const useCreateGameDtoStore = defineStore(StoreIds.CREATE_GAME_DTO, () => {
  const rolesStore = useRolesStore();
  const { getRoleWithNameInRoles } = rolesStore;

  const createGameOptionsDtoFromLocalStorage = useLocalStorage(LocalStorageKeys.GAME_OPTIONS, DEFAULT_GAME_OPTIONS, { mergeDefaults: true });

  const defaultCreateGameDto = CreateGameDto.create({
    players: [],
    options: DEFAULT_GAME_OPTIONS,
  });

  const createGameDto = ref<CreateGameDto>(CreateGameDto.create({
    ...defaultCreateGameDto,
    options: createGameOptionsDtoFromLocalStorage.value,
  }));

  const createGameOptionsDto = computed<GameOptions>(() => createGameDto.value.options);

  const doesCreateGameDtoContainPositionDependantRoles = computed<boolean>(() => {
    const positionDependantRoles: RoleName[] = ["rusty-sword-knight", "bear-tamer", "fox"];
    const playersWithPositionDependantRoles = getPlayersWithAnyRoleNameInCreateGameDto(positionDependantRoles);

    return playersWithPositionDependantRoles.length > 0;
  });

  const doesCreateGameDtoContainAdditionalCardsDependantRoles = computed<boolean>(() => {
    const playersWithAdditionalCardsDependantRoles = getPlayersWithAnyRoleNameInCreateGameDto([...ADDITIONAL_CARDS_DEPENDANT_ROLES]);

    return playersWithAdditionalCardsDependantRoles.length > 0;
  });

  function setCreateGameDto(createGameDtoValue: CreateGameDto): void {
    createGameDto.value = CreateGameDto.create(createGameDtoValue);
    saveCreateGameOptionsDtoToLocalStorage();
  }

  function resetCreateGameDto(doesRetrieveLocalStorageValues = true): void {
    if (!doesRetrieveLocalStorageValues) {
      createGameDto.value = CreateGameDto.create(defaultCreateGameDto);
      saveCreateGameOptionsDtoToLocalStorage();

      return;
    }
    createGameDto.value = CreateGameDto.create({
      ...defaultCreateGameDto,
      options: createGameOptionsDtoFromLocalStorage.value,
    });
  }

  function resetCreateGameOptionsDto(): void {
    createGameDto.value.options = GameOptions.create(DEFAULT_GAME_OPTIONS);
    saveCreateGameOptionsDtoToLocalStorage();
  }

  function saveCreateGameOptionsDtoToLocalStorage(): void {
    createGameOptionsDtoFromLocalStorage.value = createGameDto.value.options;
  }

  function removeObsoleteAdditionalCardsFromCreateGameDto(): void {
    if (!createGameDto.value.additionalCards) {
      return;
    }
    const rolesInGame = new Set(createGameDto.value.players.map(player => player.role.name));
    createGameDto.value.additionalCards = createGameDto.value.additionalCards.filter(({ recipient }) => rolesInGame.has(recipient));
  }

  function addPlayerToCreateGameDto(player: CreateGamePlayerDto): void {
    createGameDto.value.players.push(CreateGamePlayerDto.create(player));
  }

  function updatePlayerInCreateGameDto(player: CreateGamePlayerDto): void {
    const playerIndex = createGameDto.value.players.findIndex(({ name }) => name === player.name);
    if (playerIndex !== -1) {
      createGameDto.value.players.splice(playerIndex, 1, CreateGamePlayerDto.create(player));
    }
  }

  function setPlayersToCreateGameDto(players: CreateGamePlayerDto[]): void {
    createGameDto.value.players = players.map(player => CreateGamePlayerDto.create(player));
  }

  function removePlayerFromCreateGameDto(playerName: string): void {
    const playerIndex = createGameDto.value.players.findIndex(player => player.name === playerName);
    if (playerIndex !== -1) {
      createGameDto.value.players = createGameDto.value.players.toSpliced(playerIndex, 1);
    }
  }

  function isRoleInCreateGameDto(roleName: RoleName): boolean {
    return createGameDto.value.players.some(player => player.role.name === roleName);
  }

  function getPlayersWithRoleNameInCreateGameDto(roleName: RoleName): CreateGamePlayerDto[] {
    return createGameDto.value.players.filter(player => player.role.name === roleName);
  }

  function getPlayersWithAnyRoleNameInCreateGameDto(roleNames: RoleName[]): CreateGamePlayerDto[] {
    return createGameDto.value.players.filter(player => player.role.name && roleNames.includes(player.role.name));
  }

  function isRoleMinReachedInCreateGameDto(roleName: RoleName): boolean {
    const playersWithRoleName = getPlayersWithRoleNameInCreateGameDto(roleName);
    const role = getRoleWithNameInRoles(roleName);

    return !!role && (role.minInGame === undefined || playersWithRoleName.length >= role.minInGame);
  }

  function isRoleMaxReachedInCreateGameDto(roleName: RoleName): boolean {
    const playersWithRoleName = getPlayersWithRoleNameInCreateGameDto(roleName);
    const role = getRoleWithNameInRoles(roleName);

    return !!role && playersWithRoleName.length >= role.maxInGame;
  }

  function getRoleLeftCountToReachMinInCreateGameDto(roleName: RoleName): number {
    const playersWithRoleName = getPlayersWithRoleNameInCreateGameDto(roleName);
    const role = getRoleWithNameInRoles(roleName);
    if (role?.minInGame === undefined) {
      return 0;
    }
    const leftCount = role.minInGame - playersWithRoleName.length;

    return leftCount <= -1 ? 0 : leftCount;
  }

  function setAdditionalCardsForRecipientInCreateGameDto(additionalCards: CreateGameAdditionalCardDto[], recipient: GameAdditionalCardRecipientRoleName): void {
    if (!createGameDto.value.additionalCards) {
      createGameDto.value.additionalCards = additionalCards;
    }

    createGameDto.value.additionalCards = createGameDto.value.additionalCards.filter(card => card.recipient !== recipient);
    createGameDto.value.additionalCards.push(...additionalCards);
  }

  function getAdditionalCardsForRecipientInCreateGameDto(recipient: GameAdditionalCardRecipientRoleName): CreateGameAdditionalCardDto[] {
    return createGameDto.value.additionalCards?.filter(card => card.recipient === recipient) ?? [];
  }

  function getAdditionalCardsWithRoleNameInCreateGameDto(roleName: RoleName): CreateGameAdditionalCardDto[] {
    return createGameDto.value.additionalCards?.filter(card => card.roleName === roleName) ?? [];
  }
  return {
    createGameOptionsDtoFromLocalStorage,
    createGameDto,
    createGameOptionsDto,
    doesCreateGameDtoContainPositionDependantRoles,
    doesCreateGameDtoContainAdditionalCardsDependantRoles,
    setCreateGameDto,
    resetCreateGameDto,
    resetCreateGameOptionsDto,
    saveCreateGameOptionsDtoToLocalStorage,
    removeObsoleteAdditionalCardsFromCreateGameDto,
    addPlayerToCreateGameDto,
    updatePlayerInCreateGameDto,
    setPlayersToCreateGameDto,
    removePlayerFromCreateGameDto,
    isRoleInCreateGameDto,
    getPlayersWithRoleNameInCreateGameDto,
    getPlayersWithAnyRoleNameInCreateGameDto,
    isRoleMinReachedInCreateGameDto,
    isRoleMaxReachedInCreateGameDto,
    getRoleLeftCountToReachMinInCreateGameDto,
    setAdditionalCardsForRecipientInCreateGameDto,
    getAdditionalCardsForRecipientInCreateGameDto,
    getAdditionalCardsWithRoleNameInCreateGameDto,
  };
});

export { useCreateGameDtoStore };