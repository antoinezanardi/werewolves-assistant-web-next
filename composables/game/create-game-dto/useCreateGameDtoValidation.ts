import type { ComputedRef, Ref } from "vue";

import { MIN_PLAYERS_IN_GAME } from "~/composables/api/game/constants/game.constants";
import type { CreateGameDto } from "~/composables/api/game/dto/create-game/create-game.dto";

type UseCreateGameDtoValidation = {
  isMinimumPlayersReached: ComputedRef<boolean>;
  areAllRolesSet: ComputedRef<boolean>;
  canCreateGame: ComputedRef<boolean>;
};

function useCreateGameDtoValidation(createGameDto: Ref<CreateGameDto>): UseCreateGameDtoValidation {
  const isMinimumPlayersReached = computed<boolean>(() => createGameDto.value.players.length >= MIN_PLAYERS_IN_GAME);

  const areAllRolesSet = computed<boolean>(() => createGameDto.value.players.every(player => player.role.name !== undefined));

  const canCreateGame = computed<boolean>(() => isMinimumPlayersReached.value && areAllRolesSet.value);

  return {
    isMinimumPlayersReached,
    areAllRolesSet,
    canCreateGame,
  };
}

export { useCreateGameDtoValidation };