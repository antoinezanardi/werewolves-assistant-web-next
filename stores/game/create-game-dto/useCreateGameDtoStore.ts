import { defineStore } from "pinia";
import { ref } from "vue";

import { CreateGamePlayerDto } from "~/composables/api/game/dto/create-game/create-game-player/create-game-player.dto";
import { CreateGameDto } from "~/composables/api/game/dto/create-game/create-game.dto";
import { StoreIds } from "~/stores/enums/store.enum";

const useCreateGameDtoStore = defineStore(StoreIds.CREATE_GAME_DTO, () => {
  const createGameDto = ref<CreateGameDto>(CreateGameDto.create({ players: [] }));

  function setCreateGameDto(createGameDtoValue: CreateGameDto): void {
    createGameDto.value = CreateGameDto.create(createGameDtoValue);
  }

  function resetCreateGameDto(): void {
    createGameDto.value = CreateGameDto.create({ players: [] });
  }

  function addPlayerToCreateGameDto(player: CreateGamePlayerDto): void {
    createGameDto.value.players.push(CreateGamePlayerDto.create(player));
  }

  function updatePlayerInCreateGameDto(player: CreateGamePlayerDto): void {
    const playerIndex = createGameDto.value.players.findIndex(({ name }) => name === player.name);
    if (playerIndex !== -1) {
      createGameDto.value.players.splice(playerIndex, 1, CreateGamePlayerDto.create(player));
    }
  }

  function setPlayersToCreateGameDto(players: CreateGamePlayerDto[]): void {
    createGameDto.value.players = players.map(player => CreateGamePlayerDto.create(player));
  }

  function removePlayerFromCreateGameDto(playerName: string): void {
    const playerIndex = createGameDto.value.players.findIndex(player => player.name === playerName);
    if (playerIndex !== -1) {
      createGameDto.value.players = createGameDto.value.players.toSpliced(playerIndex, 1);
    }
  }
  return {
    createGameDto,
    setCreateGameDto,
    resetCreateGameDto,
    addPlayerToCreateGameDto,
    updatePlayerInCreateGameDto,
    setPlayersToCreateGameDto,
    removePlayerFromCreateGameDto,
  };
});

export { useCreateGameDtoStore };