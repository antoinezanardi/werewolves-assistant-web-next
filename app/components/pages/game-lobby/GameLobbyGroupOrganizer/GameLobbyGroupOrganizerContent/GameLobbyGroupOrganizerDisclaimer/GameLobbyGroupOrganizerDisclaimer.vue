<template>
  <div id="game-lobby-group-organizer-disclaimer">
    <Transition
      mode="out-in"
      name="fade"
    >
      <h4
        :key="groupsDisclaimerText"
        class="text-center"
      >
        <FontAwesomeIcon
          id="groups-disclaimer-icon"
          class="me-3"
          :class="groupsDisclaimerIconAndIconClasses.iconClass"
          :icon="groupsDisclaimerIconAndIconClasses.icon"
        />

        <span id="groups-disclaimer">
          {{ groupsDisclaimerText }}
        </span>
      </h4>
    </Transition>
  </div>
</template>

<script setup lang="ts">
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { storeToRefs } from "pinia";
import { MIN_PLAYERS_IN_GROUP } from "~/composables/api/game/constants/game.constants";
import type { CreateGamePlayerWithGroupDto } from "~/composables/api/game/dto/create-game/create-game-player/create-game-player.dto";
import { useCreateGameDtoValidation } from "~/composables/api/game/useCreateGameDtoValidation";
import { useCreateGameDtoStore } from "~/stores/game/create-game-dto/useCreateGameDtoStore";
import type { IconAndIconClass } from "~/utils/types/icon.types";

const createGameDtoStore = useCreateGameDtoStore();
const { firstGroupName, secondGroupName, createGameDto } = storeToRefs(createGameDtoStore);
const { getPlayersInGroupInCreateGameDto } = createGameDtoStore;

const { isMinimumPlayersReached, arePlayerGroupsSetForPrejudicedManipulatorIfPresent } = useCreateGameDtoValidation(createGameDto);

const { t } = useI18n();

const playersInFirstGroup = computed<CreateGamePlayerWithGroupDto[]>(() => getPlayersInGroupInCreateGameDto(firstGroupName.value));
const playersInSecondGroup = computed<CreateGamePlayerWithGroupDto[]>(() => getPlayersInGroupInCreateGameDto(secondGroupName.value));

const doesFirstGroupHaveEnoughPlayers = computed<boolean>(() => playersInFirstGroup.value.length >= MIN_PLAYERS_IN_GROUP);
const doesSecondGroupHaveEnoughPlayers = computed<boolean>(() => playersInSecondGroup.value.length >= MIN_PLAYERS_IN_GROUP);

const leftToFillFirstGroup = computed<number>(() => MIN_PLAYERS_IN_GROUP - playersInFirstGroup.value.length);
const leftToFillSecondGroup = computed<number>(() => MIN_PLAYERS_IN_GROUP - playersInSecondGroup.value.length);

const groupsDisclaimerText = computed<string>(() => {
  if (!isMinimumPlayersReached.value) {
    return t("components.GameLobbyGroupOrganizerDisclaimer.notEnoughPlayers");
  }
  if (!doesFirstGroupHaveEnoughPlayers.value) {
    return t("components.GameLobbyGroupOrganizerDisclaimer.fillFirstGroup", { count: leftToFillFirstGroup.value });
  }
  if (!doesSecondGroupHaveEnoughPlayers.value) {
    return t("components.GameLobbyGroupOrganizerDisclaimer.fillSecondGroup", { count: leftToFillSecondGroup.value });
  }
  return t("components.GameLobbyGroupOrganizerDisclaimer.allGroupsFilled");
});

const groupsDisclaimerIconAndIconClasses = computed<IconAndIconClass>(() => {
  if (arePlayerGroupsSetForPrejudicedManipulatorIfPresent.value) {
    return {
      icon: "check-circle",
      iconClass: "text-success",
    };
  }
  return {
    icon: "exclamation-circle",
    iconClass: "text-error fa-beat",
  };
});
</script>