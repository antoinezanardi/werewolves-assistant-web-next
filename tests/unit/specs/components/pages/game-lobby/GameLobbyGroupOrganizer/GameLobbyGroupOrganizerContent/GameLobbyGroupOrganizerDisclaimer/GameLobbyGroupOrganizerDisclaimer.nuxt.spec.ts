import type { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { createTestingPinia } from "@pinia/testing";
import { createFakeCreateGamePlayerDto } from "@tests/unit/utils/factories/composables/api/game/dto/create-game/create-game-player/create-game-player.dto.factory";
import { createFakeCreateGameDto } from "@tests/unit/utils/factories/composables/api/game/dto/create-game/create-game.dto.factory";
import type { mount } from "@vue/test-utils";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";
import GameLobbyGroupOrganizerDisclaimer from "~/components/pages/game-lobby/GameLobbyGroupOrganizer/GameLobbyGroupOrganizerContent/GameLobbyGroupOrganizerDisclaimer/GameLobbyGroupOrganizerDisclaimer.vue";
import type { CreateGameDto } from "~/composables/api/game/dto/create-game/create-game.dto";
import { StoreIds } from "~/stores/enums/store.enum";
import { useCreateGameDtoStore } from "~/stores/game/create-game-dto/useCreateGameDtoStore";

describe("Game Lobby Group Organizer Disclaimer Component", () => {
  const defaultPlayers = [
    createFakeCreateGamePlayerDto({
      name: "Antoine",
      role: { name: "prejudiced-manipulator" },
    }),
    createFakeCreateGamePlayerDto({
      name: "Bobby",
      role: { name: "werewolf" },
    }),
    createFakeCreateGamePlayerDto({
      name: "Cindy",
      role: { name: "villager" },
    }),
    createFakeCreateGamePlayerDto({
      name: "Derek",
      role: { name: "villager" },
    }),
  ];
  const defaultCreateGameDto = createFakeCreateGameDto({ players: defaultPlayers });
  const defaultFirstGroupName = "Group 1";
  const defaultSecondGroupName = "Group 2";
  const testingPinia = {
    initialState: {
      [StoreIds.CREATE_GAME_DTO]: {
        createGameDto: defaultCreateGameDto,
        firstGroupName: defaultFirstGroupName,
        secondGroupName: defaultSecondGroupName,
      },
    },
    stubActions: false,
  };
  let wrapper: ReturnType<typeof mount<typeof GameLobbyGroupOrganizerDisclaimer>>;

  async function mountGameLobbyGroupOrganizerDisclaimerComponent(options: ComponentMountingOptions<typeof GameLobbyGroupOrganizerDisclaimer> = {}):
  Promise<ReturnType<typeof mount<typeof GameLobbyGroupOrganizerDisclaimer>>> {
    return mountSuspendedComponent(GameLobbyGroupOrganizerDisclaimer, {
      global: { plugins: [createTestingPinia(testingPinia)] },
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountGameLobbyGroupOrganizerDisclaimerComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Groups Disclaimer and Icon", () => {
    it.each<{
      test: string;
      createGameDto: CreateGameDto;
      expectedIcon: string;
      expectedIconClass: string;
      expectedDisclaimer: string;
    }>([
      {
        test: "should render the exclamation circle icon with text error and disclaimer not enough players when the create game dto doesn't have enough players.",
        createGameDto: createFakeCreateGameDto({
          players: [defaultPlayers[0]],
        }),
        expectedIcon: "exclamation-circle",
        expectedIconClass: "text-error",
        expectedDisclaimer: "components.GameLobbyGroupOrganizerDisclaimer.notEnoughPlayers",
      },
      {
        test: "should render the exclamation circle icon with text error and disclaimer not enough groups in first group when there is not enough players in first group.",
        createGameDto: createFakeCreateGameDto({
          players: [
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[0],
              group: defaultFirstGroupName,
            }),
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[1],
              group: defaultSecondGroupName,
            }),
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[2],
              group: defaultSecondGroupName,
            }),
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[3],
              group: defaultSecondGroupName,
            }),
          ],
        }),
        expectedIcon: "exclamation-circle",
        expectedIconClass: "text-error",
        expectedDisclaimer: `components.GameLobbyGroupOrganizerDisclaimer.fillFirstGroup, {"count":1}`,
      },
      {
        test: "should render the exclamation circle icon with text error and disclaimer not enough groups in second group when there is not enough players in second group.",
        createGameDto: createFakeCreateGameDto({
          players: [
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[0],
              group: defaultFirstGroupName,
            }),
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[1],
              group: defaultFirstGroupName,
            }),
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[2],
              group: defaultFirstGroupName,
            }),
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[3],
              group: defaultSecondGroupName,
            }),
          ],
        }),
        expectedIcon: "exclamation-circle",
        expectedIconClass: "text-error",
        expectedDisclaimer: `components.GameLobbyGroupOrganizerDisclaimer.fillSecondGroup, {"count":1}`,
      },
      {
        test: "should render the check circle icon with text success and valid groups disclaimer when groups are valid.",
        createGameDto: createFakeCreateGameDto({
          players: [
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[0],
              group: defaultFirstGroupName,
            }),
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[1],
              group: defaultFirstGroupName,
            }),
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[2],
              group: defaultSecondGroupName,
            }),
            createFakeCreateGamePlayerDto({
              ...defaultPlayers[3],
              group: defaultSecondGroupName,
            }),
          ],
        }),
        expectedIcon: "check-circle",
        expectedIconClass: "text-success",
        expectedDisclaimer: "components.GameLobbyGroupOrganizerDisclaimer.allGroupsFilled",
      },
    ])("$test", async({ createGameDto, expectedIcon, expectedIconClass, expectedDisclaimer }) => {
      const createGameDtoStore = useCreateGameDtoStore();
      createGameDtoStore.createGameDto = createFakeCreateGameDto(createGameDto);
      await nextTick();
      const icon = wrapper.findComponent<typeof FontAwesomeIcon>("#groups-disclaimer-icon");
      const disclaimer = wrapper.find<HTMLSpanElement>("#groups-disclaimer");

      expect(icon.classes()).toContain(expectedIconClass);
      expect(icon.props("icon")).toBe(expectedIcon);
      expect(disclaimer.text()).toBe(expectedDisclaimer);
    });
  });
});