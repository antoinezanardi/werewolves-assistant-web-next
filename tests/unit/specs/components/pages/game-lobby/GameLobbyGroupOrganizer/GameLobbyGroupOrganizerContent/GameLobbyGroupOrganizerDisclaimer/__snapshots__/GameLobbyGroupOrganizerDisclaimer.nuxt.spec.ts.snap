// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game Lobby Group Organizer Disclaimer Component > should match snapshot when rendered. 1`] = `
"<div id="game-lobby-group-organizer-disclaimer">
  <transition-stub mode="out-in" name="fade" appear="false" persisted="false" css="true">
    <h4 class="text-center">
      <font-awesome-icon-stub icon="exclamation-circle" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" id="groups-disclaimer-icon" class="me-3 text-error fa-beat"></font-awesome-icon-stub><span id="groups-disclaimer">components.GameLobbyGroupOrganizerDisclaimer.fillFirstGroup, {"count":2}</span>
    </h4>
  </transition-stub>
</div>"
`;
